### PR TITLE
Nullable: System.Runtime.InteropServices.Marshal

### PR DESCRIPTION
**Pull request by tannergooding**
CC. @dotnet/nullablefc 

Putting up `Marshal` by itself, since it is a bit more involved.

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs [hunk 1]
```diff
@@ -2,14 +2,15 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs]: This file should match `Marshal.CoreCLR.cs`, where the APIs are the same.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs [hunk 2]
```diff
@@ -110,7 +108,8 @@ public static unsafe string PtrToStringUTF8(IntPtr ptr)
             return string.CreateStringFromEncoding((byte*)ptr, nbBytes, Encoding.UTF8);
         }
 
-        public static unsafe string PtrToStringUTF8(IntPtr ptr, int byteLen)
+        // TODO-NULLABLE: This has different behavior from the other PtrToString(IntPtr, int) functions
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: The other APIs return `null` for `ptr == IntPtr.Zero` on the overload that doesn't take a length; and throw for the input that does take a length.

We should probably make these consistent.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs [hunk 3]
```diff
@@ -618,21 +618,21 @@ public static void ThrowExceptionForHR(int errorCode)
         {
             if (errorCode < 0)
             {
-                throw GetExceptionForHR(errorCode, IntPtr.Zero);
+                throw GetExceptionForHR(errorCode, IntPtr.Zero)!;
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: The compiler complains for `throw null`... Maybe it shouldn't since that is valid and transforms to `throw NullReferenceException`?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs [hunk 4]
```diff
@@ -618,21 +618,21 @@ public static void ThrowExceptionForHR(int errorCode)
...
             }
         }
 
         public static void ThrowExceptionForHR(int errorCode, IntPtr errorInfo)
         {
             if (errorCode < 0)
             {
-                throw GetExceptionForHR(errorCode, errorInfo);
+                throw GetExceptionForHR(errorCode, errorInfo)!;
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: This might be an interesting scenario in that we know the result isn't null here; since it is only `null` for positive errorCode.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs [hunk 5]
```diff
@@ -371,7 +367,7 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
         private static extern IntPtr /* IUnknown* */ GetComInterfaceForObjectNative(object o, Type t, bool onlyInContext, bool fEnalbeCustomizedQueryInterface);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern object GetObjectForIUnknown(IntPtr /* IUnknown* */ pUnk);
+        public static extern object? GetObjectForIUnknown(IntPtr /* IUnknown* */ pUnk);
```
Comment (tannergooding) [inline src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: As best as I could tell; the native code supports returning `null`. But, I'm unsure if you can actually give a valid `IntPtr` that would result in that ever happening... Same applies to most of the other functions with a nullable return below.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs [hunk 6]
```diff
@@ -37,20 +38,22 @@ public static void CleanupUnusedObjectsInCurrentContext()
...
 
-        public static object CreateWrapperOfType(object o, Type t)
+        public static object? CreateWrapperOfType(object? o, Type t)
         {
             throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
         }
 
         public static TWrapper CreateWrapperOfType<T, TWrapper>(T o)
         {
+            // TODO-NULLABLE: Generic T can be null
```
Comment (krwq) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs]: I think this can be represented with `where T: class?`

Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs]: That looks to add a `where T : class` constraint, which changes the signature

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs]: Yes, we would need `[MaybeNull]` or `where T : object?`.

We've been tagging these with `// TODO-NULLABLE-GENERIC

Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs]: @stephentoub would you prefer I change it to: `// TODO-NULLABLE-GENERIC: T can be null`? I think the additional note on whether T can or cannot be null is useful

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs]: No preference... but pretty much all of the cases where we have TODO-NULLABLE-GENERIC are in the same camp.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 2 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: CC. @jkoritzinsky.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs (hunk 6 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs]: Fixed.

File: src/System.Private.CoreLib/shared/Interop/Unix/System.Native/Interop.GetHostName.cs [hunk 7]
```diff
@@ -35,7 +35,7 @@ internal static unsafe string GetHostName()
             // If the hostname is truncated, it is unspecified whether the returned buffer includes a terminating null byte.
             name[ArrLength - 1] = 0;
 
-            return Marshal.PtrToStringAnsi((IntPtr)name);
+            return Marshal.PtrToStringAnsi((IntPtr)name)!;
```
Comment (tannergooding) [inline src/System.Private.CoreLib/shared/Interop/Unix/System.Native/Interop.GetHostName.cs]: `Marshal.PtrToStringAnsi` only returns `null` if the parameter is `IntPtr.Zero` or a Win32 atom, so this shouldn't be `null` here based on what `name` can be (same goes for `GetUnixName` and `GetAllMountPoints` below).

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs [hunk 8]
```diff
@@ -232,14 +231,14 @@ public static void Copy(IntPtr[] source, int startIndex, IntPtr destination, int
...
-            if (source == null)
+            if (source is null)
                 throw new ArgumentNullException(nameof(source));
             if (destination == IntPtr.Zero)
                 throw new ArgumentNullException(nameof(destination));
 
             // The rest of the argument validation is done by CopyTo
 
-            new Span<T>(source, startIndex, length).CopyTo(new Span<T>((void*)destination, length));
+            source.AsSpan(startIndex, length).CopyTo(new Span<T>((void*)destination, length));
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Why did you change this?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs [hunk 9]
```diff
@@ -295,7 +294,7 @@ private static unsafe void CopyToManaged<T>(IntPtr source, T[] destination, int
 
             // The rest of the argument validation is done by CopyTo
 
-            new Span<T>((void*)source, length).CopyTo(new Span<T>(destination, startIndex, length));
+            new Span<T>((void*)source, length).CopyTo(destination.AsSpan(startIndex, length));
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Same question... they're functionally identical, as AsSpan just calls the ctor, but using AsSpan just forces the JIT to do more inlining.  Is there a benefit?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs [hunk 10]
```diff
@@ -532,38 +531,37 @@ public static void Prelink(MethodInfo m)
...
         {
-            if (c == null)
+            if (c is null)
             {
                 throw new ArgumentNullException(nameof(c));
             }
 
             MethodInfo[] mi = c.GetMethods();
-            if (mi != null)
+
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: This is ok because no override of GetMethods will return null?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 3 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: > The compiler complains

Sounds like an issue to be opened in the Roslyn repo?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 4 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: > This might be an interesting scenario in that we know the result isn't null here; since it is only null for positive errorCode.

Consider adding a suggested attribute to https://github.com/dotnet/roslyn/issues/26761.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: > We should probably make these consistent.

Can you open a separate issue for that?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 8 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Ah, I was prototyping an extension and forgot to disable it on my home machine. I'll revert.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 10 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Right, they shouldn't. The documentation for the API is:
> An array of MethodInfo objects representing all methods defined for the current Type that match the specified binding constraints.
>
> -or-
>
> An empty array of type MethodInfo, if no methods are defined for the current Type, or if none of the defined methods match the binding constraints.

This should also match the concurrent PR against Type.

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Hmm.  @jkotas, any opinion on this?

Even if the docs say you shouldn't do this, and even if our annotations will say it doesn't return null, and even though this is super rare, since we've always previously checked and since it is technically possible for someone to override this and return null, and since this doesn't seem like a particularly hot path we need to eliminate branches on, I'd be inclined to leave the null check in place.

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: I am fine with deleting the null check. Nobody will notice.

This is useless API with exactly zero use: https://apisof.net/catalog/System.Runtime.InteropServices.Marshal.PrelinkAll(Type) . The only reason why it exists in .NET Core is a check box for NS2.0.

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: > This is useless API with exactly zero use … The only reason why it exists in .NET Core is a check box for NS2.0

The gift that keeps on giving :)

> I am fine with deleting the null check. Nobody will notice.

Ok, sounds good.  Thoughts in general on this?  What if it were a more commonly used method?

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Also, given that, why not just neuter the method entirely?

Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: > Ok, sounds good. Thoughts in general on this? What if it were a more commonly used method?

My general thought here is that; just from reading the docs (and completely ignoring the nullable annotations/feature); I would have no reason to believe I should need a null check here and would likely consume `Type.GetMethods()` without the null check.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 2 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Logged https://github.com/dotnet/coreclr/issues/23723

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 10 above)
Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: > What if it were a more commonly used method?

Case-by-case. If this null check was in a general purpose reflection helper that the code is likely going to pass a custom reflection objects into, I would keep it. Out of curiosity, I would also looked at history to see whether the null check was added as a bug fix, or whether it was just a defensive programming to start with. https://github.com/dotnet/coreclr/pull/20252 is an example of such superfluous null check that is worth keeping.

> why not just neuter the method entirely?

There may be somebody somewhere using this method for what it does...

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 3 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Logged https://github.com/dotnet/roslyn/issues/34754

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 4 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Done.

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs (hunk 10 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: >  Out of curiosity, I would also looked at history to see whether the null check was added as a bug fix, or whether it was just a defensive programming to start with. 

It was included as part of the initial CoreCLR code population and I'm not quite sure the appropriate place to start digging deeper for the TFS sources that far back.

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Most of the history is in SourceDepot. I have checked this one and this null check was there in 2001 (2001 is as far as the history in SourceDepot goes).

Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: I'm fine with either keeping it or removing it. But, I feel that if we keep it we should likely update the documentation to indicate that someone could potentially return `null` here; as most developers are going to follow what the documentation says for things like `Type.GetMethods()`.

I also imagine that, regardless of `nullable annotations` not providing a "guarantee" that something won't return null (even if annotated that way); people are going to start relying on these annotations and removing checks from their own code if the API says it won't return `null` and logging bugs if it turns out the API is not annotated correctly.

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs]: Keep the null check removed here and be done with it. It is not worth to spend any more time on this specific case.

`Type.GetMethods()` should be annotated as returning non-null, to match the documented contract.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs [hunk 11]
```diff
@@ -380,21 +376,22 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
         /// ReleaseComObject on a RCW and not worry about other active uses ofsaid RCW.
         /// </summary>
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern object GetUniqueObjectForIUnknown(IntPtr unknown);
+        public static extern object? GetUniqueObjectForIUnknown(IntPtr unknown);
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: This does not look right. When this can return null?

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs [hunk 12]
```diff
@@ -380,21 +376,22 @@ public static string GetTypeInfoName(ITypeInfo typeInfo)
...
 
         /// <summary>
         /// Return an Object for IUnknown, using the Type T.
         /// Type T should be either a COM imported Type or a sub-type of COM imported Type
         /// </summary>
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern object GetTypedObjectForIUnknown(IntPtr /* IUnknown* */ pUnk, Type t);
+        public static extern object? GetTypedObjectForIUnknown(IntPtr /* IUnknown* */ pUnk, Type t);
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: Same here

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs [hunk 13]
```diff
@@ -713,12 +710,12 @@ public static T[] GetObjectsForNativeVariants<T>(IntPtr aSrcNativeVariant, int c
         [MethodImpl(MethodImplOptions.InternalCall)]
         public static extern int GetEndComSlot(Type t);
 
-        public static object BindToMoniker(string monikerName)
+        public static object? BindToMoniker(string monikerName)
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: When can this return null?

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs (hunk 5 above)
Comment (jkotas) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: > the native code supports returning null

Where do you see that? I do not think that this can ever return null.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs (hunk 11 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: I called this out here: https://github.com/dotnet/coreclr/pull/23699#discussion_r271908202

It looks like `MarshalNative::GetUniqueObjectForIUnknown` just forwards to `GetObjectRefFromComIP`. `GetObjectRefFromComIP` then doesn't look to ensure that `pObjOut` is non-null for every potential code path; but I was unsure if that can actually happen in practice and it looked like for the `GetUniqueObjectForIUnknown` case, it should always try to find or create an object ref if one doesn't exist yet...

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs (hunk 13 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: Ah, I missed that the P/Invoke methods were `PreserveSig = false`; so they will throw if any of them fail. I'll fix this.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs (hunk 11 above)
Comment (jkotas) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: Right. If you pass non-null punk to `GetObjectRefFromComIP. GetObjectRefFromComIP`, it will give you non-null object back.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs (hunk 5 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: I believe it's just a potential case, but `GetObjectRefFromComIP` will definitely assign `pObjOut` to `NULL` if `*ppUnk == NULL` (which shouldn't happen from our managed entry points). Otherwise, it looks like it tries to always find or create a wrapper for the given interface.

I also believe it should fail if you pass in a non-null `IntPtr` which is not an `IUnknown`; but I didn't dig that deeply into it.

I called it out here to ensure I got validation from the people who have more knowledge of CCW/RCW before the PR got merged; and I'll change these to be non-nullable.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs (hunk 13 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: Fixed.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs (hunk 12 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: Fixed.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs (hunk 11 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: Fixed.

File: src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs [hunk 14]
```diff
@@ -668,34 +666,33 @@ public static object CreateWrapperOfType(object o, Type t)
...
         public static void GetNativeVariantForObject<T>(T obj, IntPtr pDstNativeVariant)
         {
-            GetNativeVariantForObject((object)obj, pDstNativeVariant);
+            // TODO-NULLABLE-GENERIC: T can be null
+            GetNativeVariantForObject((object)obj!, pDstNativeVariant);
         }
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern object GetObjectForNativeVariant(/* VARIANT * */ IntPtr pSrcNativeVariant);
+        public static extern object? GetObjectForNativeVariant(/* VARIANT * */ IntPtr pSrcNativeVariant);
```
Comment (tannergooding) [inline src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs]: This can return `null` since `MarshalNative::GetObjectForNativeVariant` forwards to `OleVariant::MarshalObjectForOleVariant` which will return null for `VT_EMPTY`

**Comment by tannergooding** [comment]
@stephentoub, do you have any other feedback here?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Unix.cs [hunk 15]
```diff
@@ -2,29 +2,32 @@
...
 using System.Diagnostics;
 using System.Text;
 
 namespace System.Runtime.InteropServices
 {
     public static partial class Marshal
     {
-        public static string PtrToStringAuto(IntPtr ptr, int len)
+        // TODO-NULLABLE: This has different behavior from the other PtrToString(IntPtr, int) functions
+        //                This is due to PtrToStringUTF8 on Unix and is being resolved independently
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Unix.cs]: Do we still need the TODO-NULLABLE then?  Or can you include the issue #?

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Windows.cs [hunk 16]
```diff
@@ -2,28 +2,31 @@
...
+#nullable enable
 using System.Diagnostics;
 
 namespace System.Runtime.InteropServices
 {
     public static partial class Marshal
     {
-        public static string PtrToStringAuto(IntPtr ptr, int len)
+        // TODO-NULLABLE: This has different behavior from the other PtrToString(IntPtr, int) functions
+        //                This is due to PtrToStringUTF8 on Unix and is being resolved independently
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Windows.cs]: Ditto

File: src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Unix.cs (hunk 15 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.Unix.cs]: I could probably update the comment, but it likely isn't worth the extra CI cycles. I'm just waiting on https://github.com/dotnet/coreclr/pull/23731 to be merged and for us to pull it back into NullableFeature (probably resolved Monday).
